### DISCUSSION TITLE
GraphQL API: Support for updating checkout line quantity by variantId

### DISCUSSION THREAD

**Discussion by akhial** [saleor/saleor#10685] [Ideas]
Hi Saleor team!

Saleor 3.6 has deprecated setting lines' quantities by `variantId` in mutation `checkoutLinesUpdate`, this feature was useful as a way to reduce variant quantities in a checkout line. Is the alternative way the following?

1- Query `checkout` by `id` and filter `Checkout.lines` on client side for `lineId` that matches `variantId`
2- Call `checkoutLinesUpdate` with input `lineId` and modify the quantity

Previous method was:

1- Call `checkoutLinesUpdate` with input `variantId` and modify the quantity

#### Value of previous method
* Support for reducing checkout quantities in one request instead of two
* Removed complexity of tracking cart quantity changes client-side, as we could just send the new quantity instead of needing to figure out the number of newly added items; Which would then be used in `checkoutLinesAdd`.

#### Proposal
For the above reasons I believe it would be better for API consumer experience to keep the overwrite quantity by `variantId` behavior somehow... perhaps with a `overwriteQuantity: Boolean` in `CheckoutLineInput` for `checkoutLinesAdd`?

Adel

**Comment by IKarbowiak** [comment]
Hi @akhial,

We introduced the feature that allows creating multiple lines for the same variant, because of that updating the lines cannot be performed just by `variantId` anymore, as we missed the information which line should be updated. I hope it clarifies to you what was the reason behind this change.
